cmake/Modules/Packages/KOKKOS.cmake

```diff
@@ -1,6 +1,8 @@
 ########################################################################
 # As of version 3.3.0 Kokkos requires C++14
-set(CMAKE_CXX_STANDARD 14)
+if(CMAKE_CXX_STANDARD LESS 14)
```

Comment:
@junghans @rbberger this is a "quick hack workaround" for the stable release, but not a satisfactory solution. It seems that when globally the standard is set to c++11 some files are compiled with -std=c++11 and then the remainder with -std=c++14. This only is to avoid having this to lower the standard required globally. So probably this statement would have to come where the standard is set globally and also bracketed by "if(PKG_KOKKOS)". Is there some other way in CMake to change a "global" setting retroactively from within some some included script code?

Reply:
Is it a problem to mix C++11 lammps with C++14 kokkos?

Reply:
I don't expect that there is a problem, but it is an inconsistent behavior because C++11 is applied to all sources "processed" before the standard is switched by KOKKOS.cmake. 

On top of that, all C++ code linked together (i.e. using the same headers) is supposed to be compiled with consistent settings with respect to C++ (e.g. exceptions and rtti and alike), so by switching midway this "rule" is violated (which also triggers concerns about including external packages like an external Kokkos installation. Those likely will have to be compiled and use with consistent settings). Now I don't think that is a problem in practice (yet), but more sweeping changes, like those introduced by C++20 may be more sensitive to that. So effectively it looks to me as if adding the if statement here helps by not lowering the standard for files where the later standard is expected, but if LAMMPS requires "only" C++11 it may cause undefined behavior (probably not for C++11 vs C++14 but perhaps if Kokkos or some other future package would require a much later standard).

In short, for the stable release this is acceptable, but in general, some thinking may be needed. The same concern about consistent settings would also apply to the case of using and external project versus fetch content and adding the subdirectory like in the multi-config PR. When I read up on multi-config build systems, there was significant discussions about having the same config setting (Debug, Release, etc.) for all "modules" of a compilation in the same sense as I mentioned above.

Reply:
I see, that is because we `include(KOKKOS.cmake)` instead of using a local CMake scope like a subdirectory, function or target property, I guess we could apply C++14 only on the kokkos files.

Your change here looks ok, but I guess setting the standard in the main CMake file might be more obvious even if  it is inside a `if(PKG_KOKKOS)` block.

Reply:
That is exactly my thinking and the change I am preparing for later. I think overall we'll be better off to have the same C++ standard setting for all files, even if it puts some extra effort on testing and how to write code.